atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/HostnamesByIpSupplierTest.java

```diff
@@ -93,17 +93,7 @@ public void returnsEmptyOnTimeout() throws Exception {
             Thread.sleep(1);
             throw new NotFoundException();
         });
-        setupKeyspaceAndTable(secondaryClient);
-        CqlResult cqlResult = createMockCqlResult(ImmutableList.of(
-                ImmutableList.of(
-                        createColumn("ip", PtBytes.toBytes("10.0.0.0")),
-                        createColumn("hostname", PtBytes.toBytes("cassandra-1"))),
-                ImmutableList.of(
-                        createColumn("ip", PtBytes.toBytes("10.0.0.1")),
-                        createColumn("hostname", PtBytes.toBytes("cassandra-2")))));
-        Mockito.lenient()
-                .when(secondaryClient.execute_cql3_query(any(), any(), any()))
-                .thenReturn(cqlResult);
```

Comment:
I think the whole point of this logic was to make sure we did not iterate the second client. However, we can easily verify that by checking if the function is called on the second client. Would be great if you can double check.

Reply:
I agree that we could double check this way. But I think the semantic meaning of the test would be different? As in, we're not really testing if we interact or not with the secondary client. We're testing that, even if the secondary client returns something, we shouldn't answer it if we end up running into a timeout. And that's very different from not interacting with the secondary client (although both of them would be true in this case, because we just go through them in a for loop and check the timer at each time). 

Also, if you want to keep the change you made, your `verify` would have to be after the `supplier.get()` is called. Otherwise nothing really has happened on the test yet, right?

I'd keep the test as it was originally.